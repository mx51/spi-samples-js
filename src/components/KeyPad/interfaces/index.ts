export interface IKeyPadProps {
  title: string;
  subtitle: string;
  defaultAmount: number;
  onAmountChange: (amount: number) => void;
  open: boolean;
  onClose: () => void;
}
