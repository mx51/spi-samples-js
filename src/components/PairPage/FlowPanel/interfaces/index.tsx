export interface IFlowPanel {
  flow: boolean;
}
