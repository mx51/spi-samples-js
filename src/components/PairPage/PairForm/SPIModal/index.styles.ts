import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    spiModalHeading: {
      fontSize: '1.125rem',
      marginBottom: -theme.spacing(1),
    },
    spiModalSubHeading: {
      color: theme.palette.text.secondary,
      fontSize: '0.875rem',
      fontWeight: 300,
      margin: theme.spacing(0, 3, 2, 3),
    },
  })
);

export default useStyles;
