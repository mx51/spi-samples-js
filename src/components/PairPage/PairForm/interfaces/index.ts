export interface ISPIModel {
  modalToggle: boolean;
  handleProviderChange: (value: string) => void;
  onClose: (newValue: string) => void;
  providerValue: string;
}

export interface IPreventDefault {
  preventDefault: () => void;
}

export interface ISPIFormData {
  provider: {
    modalToggle: boolean;
    value: string;
    isValid: boolean;
  };
  configuration: {
    type: string;
    value: string;
  };
  serialNumber: {
    value: string;
    isValid: boolean;
  };
  posId: {
    value: string;
    isValid: boolean;
  };
  apikey: {
    value: string;
    isValid: boolean;
  };
  testMode: boolean;
}

export interface ISPIAttribute {
  [attribute: string]: Record<string, unknown>;
}

export interface ISPIUpdatedData {
  value?: string;
  modalToggle?: boolean;
  isValid?: boolean;
  type?: string;
  testMode?: boolean;
}

export interface ISPIFormDataValidator {
  providerValid: boolean;
  serialNumberValid: boolean;
  posIdValid: boolean;
}

export type IFormEventValue = React.ChangeEvent<{ value: unknown }>;

export type IFormEventCheckbox = React.ChangeEvent<{ checked: boolean }>;

export type IFormEventKey = React.KeyboardEvent<{ key: string }>;
