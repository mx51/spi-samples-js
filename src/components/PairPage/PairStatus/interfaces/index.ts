export interface PairPanelButtonsInterface {
  statusTitle: string;
  statusIcon: React.ReactNode;
  statusText: string;
  button: React.ReactNode;
}

export interface PairPanelInformationInterface {
  title: string;
  content: string;
}

export interface PairStatusInterface {
  open: boolean;
  handleDrawerToggle: () => void;
}

export interface PairStatusStatesInterface {
  merchantId: string;
  terminalId: string;
  battery: string;
}
