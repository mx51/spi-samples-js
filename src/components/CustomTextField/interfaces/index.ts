import { IFormEventValue } from '../../PairPage/PairForm/interfaces';

type Partial<T> = {
  [P in keyof T]?: T[P];
};

type InputProps = {
  endAdornment?: React.ReactNode;
};

type TInputProps = {
  maxLength?: number;
};

export interface ICustomTextField {
  disabled?: boolean;
  error?: boolean;
  fullWidth?: boolean;
  helperText?: string;
  id?: string;
  inputProps?: Partial<TInputProps>;
  InputProps?: Partial<InputProps>;
  label?: string;
  margin?: 'dense' | 'none';
  onBlur?: (event: IFormEventValue) => void;
  onChange?: (event: IFormEventValue) => void;
  required?: boolean;
  value?: string;
  variant?: 'filled' | 'standard' | 'outlined' | undefined;
}

export interface IErrorInputAdornment {
  isValid: boolean;
}
