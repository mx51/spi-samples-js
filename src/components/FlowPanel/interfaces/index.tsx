import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';

export interface IFlowPanel {
  flow: boolean;
  receiptToggle?: boolean;
}

export interface IFlowPanelState {
  flow: boolean;
  terminal?: ITerminalProps | null;
}
