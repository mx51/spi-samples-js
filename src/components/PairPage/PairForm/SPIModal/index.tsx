import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import Typography from '@material-ui/core/Typography';
// constants
import {
  TEXT_FORM_MODAL_CODE_TILL,
  TEXT_FORM_MODAL_CODE_WESTPAC,
} from '../../../../definitions/constants/commonConfigs';
// styles
import useStyles from './index.styles';
// interfaces
import { ISPIModel } from '../interfaces';

function SPIModal({ modalToggle, handleProviderChange, onClose, providerValue }: ISPIModel): React.ReactElement {
  const classes = useStyles();
  const [selectedProvider, setProvider] = useState(providerValue);

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setProvider((event.target as HTMLInputElement).value);
  };
  const handleClose = () => {
    setProvider(providerValue); // keep previous selection
    onClose(providerValue);
  };
  const handleOk = () => {
    handleProviderChange(selectedProvider); // update selection;
    onClose(selectedProvider);
  };

  return (
    <Dialog aria-labelledby="spiModalTitle" open={modalToggle}>
      <DialogTitle id="spiModalTitle" className={classes.spiModalHeading}>
        Simple Payments Integration
      </DialogTitle>
      <Typography variant="body2" className={classes.spiModalSubHeading}>
        Select your Payment provider
      </Typography>
      <DialogContent dividers>
        <FormControl component="fieldset">
          <RadioGroup
            aria-label="paymentProviderSelector"
            name="paymentProviderSelector"
            value={selectedProvider}
            onChange={handleChange}
          >
            <FormControlLabel
              value={TEXT_FORM_MODAL_CODE_TILL}
              control={<Radio color="primary" />}
              label="Till Payments"
            />
            <FormControlLabel
              value={TEXT_FORM_MODAL_CODE_WESTPAC}
              control={<Radio color="primary" />}
              label="Westpac"
            />
          </RadioGroup>
        </FormControl>
      </DialogContent>
      <DialogActions>
        <Button color="primary" onClick={handleClose}>
          Cancel
        </Button>
        <Button color="primary" onClick={handleOk}>
          OK
        </Button>
      </DialogActions>
    </Dialog>
  );
}

export default SPIModal;
