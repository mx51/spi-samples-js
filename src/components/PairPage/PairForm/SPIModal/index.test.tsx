import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { cleanup, fireEvent } from '@testing-library/react';
import SPIModal from '.';
import { IPairFormParams } from '../../../../redux/reducers/PairFormSlice/interfaces';
import mockWithRedux, { defaultMockPairFormParams } from '../../../../utils/tests/common';

function setupContainer(pairForm: IPairFormParams = defaultMockPairFormParams) {
  const customizedStore = {
    getState: () => ({
      pairForm,
    }),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
  };

  return mockWithRedux(
    <SPIModal
      modalToggle
      handleProviderChange={jest.fn()}
      onClose={jest.fn()}
      providerValue={pairForm.acquirerCode.value}
    />,
    customizedStore
  );
}

describe('Test <SPIModal />', () => {
  let mockContainer: Any;

  beforeEach(() => {
    mockContainer = setupContainer();
  });

  afterEach(cleanup);

  test('should match SPIModal snapshot test', () => {
    // Assert
    expect(mockContainer).toMatchSnapshot();
  });

  test('should display modal title', () => {
    // In terms of React Portal, the modal dialog is not visible in container dom, but which can be found within document DOM tree
    // Arrange
    const title = 'Simple Payments Integration';

    // Assert
    expect(document.body.innerHTML.includes(title)).toBeTruthy();
  });

  test('should show pre-selected radio button', () => {
    // Arrange
    const checkedClassName = 'Mui-checked';
    const wbcRadioBtnDOM = document.body.querySelector('[data-test-id="wbc"]') as Element;

    // Act
    fireEvent.click(wbcRadioBtnDOM);

    // Assert
    expect(wbcRadioBtnDOM.innerHTML.includes(checkedClassName)).toBeTruthy();
  });

  test('should show the updated selected radio button', () => {
    // Arrange
    const updatedFormParams = {
      ...defaultMockPairFormParams,
      acquirerCode: {
        ...defaultMockPairFormParams.acquirerCode,
        value: 'other',
      },
    };
    const checkedClassName = 'Mui-checked';
    const checkedKeyword = 'checked';

    // Act
    setupContainer(updatedFormParams);

    const otherOptionDOM = document.body.querySelector('[data-test-id="other"]');
    const wbcOptionDOM = document.body.querySelector('[data-test-id="wbc"]');

    // Assert
    expect(otherOptionDOM?.innerHTML.includes(checkedClassName)).toBeTruthy();
    expect(wbcOptionDOM?.innerHTML.includes(checkedKeyword)).toBeFalsy();
  });
});
