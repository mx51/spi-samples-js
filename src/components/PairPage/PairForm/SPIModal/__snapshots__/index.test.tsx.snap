// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <SPIModal /> should match SPIModal snapshot test 1`] = `
<div
  aria-hidden="true"
/>
`;
